### PR TITLE
perf: improve rendering performance of mobile sidebar

### LINKED ISSUE #5470: Optimize mobile sidebar open/close rendering
### Overview

Enhance rendering performance for low-compute mobile devices, specifically focusing on the sidebar opening and closing rendering performance on mobile devices.

Performance bottlenecks are yet to be identified.

### References and additional details

1. Profile the application using Chrome DevTools with 6x throttling to identify components causing re-renders.
2. Review Chakra UI components for potential performance optimizations, such as lazy-loading or conditional rendering.
3. Implement performance optimizations to rendering performance accordingly

https://www.notion.so/shapeshift/Mobile-profiling-0321f42373a14e399db7e5255676a97c

### Acceptance Criteria

- Improved in rendering performance of sidebar open/close on low-compute devices.
- No degradation in user experience or functionality.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Enhances rendering performance for low-compute mobile devices, specifically focusing on the sidebar opening and closing rendering performance on mobile devices.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #5470

## Risk

Low risk of broken UI in mobile layout

## Testing

You can test these changes with chrome devtools rather than waiting for a mobile build:
image

- Improved in rendering performance of sidebar open/close on low-compute devices.
- No degradation in user experience or functionality.

## Screenshots (if applicable)

Sidebar opening before improvements (6x throttling on m2 mac) - 0.95s
image

Sidebar opening after improvements (6x throttling on m2 mac) - 0.47s
image

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5471** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5471" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5471?utm_source=stack-comment).

File: src/index.tsx [hunk 1]
```diff
@@ -13,11 +13,11 @@ import * as serviceWorkerRegistration from './serviceWorkerRegistration'
...
-  <React.StrictMode>
-    <AppProviders>
-      <App />
-    </AppProviders>
-  </React.StrictMode>,
+  // <React.StrictMode>
+  <AppProviders>
+    <App />
+  </AppProviders>,
+  // </React.StrictMode>,
```
Comment (woodenfurniture) [inline src/index.tsx:20]:
```suggestion
  <React.StrictMode>
    <AppProviders>
      <App />
    </AppProviders>,
  </React.StrictMode>,
```

File: src/components/Layout/Header/NavBar/Notifications.tsx [hunk 2]
```diff
@@ -119,22 +129,17 @@ export const Notifications = () => {
...
-          signMessage,
-          signTypedData,
-        }}
+        customSigner={customSignerProp}
         partnerKey={partnerKey}
         theme={themeObj}
         disableAnalytics={disableAnalytics}
       >
-        <NotificationFeed gapFromBell={10} placement={'bottom-end'}>
+        <NotificationFeed gapFromBell={10} placement='bottom-end'>
```
Comment (gomesalexandre) [inline src/components/Layout/Header/NavBar/Notifications.tsx:137]: 🧠

**Review by gomesalexandre — COMMENTED** [review]
Conceptual stamp pending runtime testing, all the memo 🎉

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Tested locally, `<UserMenu />` is broken in this build and doesn't show up anymore

image

Mobile sidenav is looking good however:

image

File: src/components/Layout/Header/Header.tsx [hunk 3]
```diff
@@ -186,39 +201,41 @@ export const Header = memo(() => {
...
               rowGap={4}
               columnGap={2}
             >
               <GlobalSeachButton />
-              {isDegradedState && <DegradedStateBanner />}
-              {isWalletConnectToDappsV2Enabled && (
-                <Box display={{ base: 'none', md: 'block' }}>
+              {isLargerThanMd && isDegradedState && <DegradedStateBanner />}
+              {isLargerThanMd && isWalletConnectToDappsV2Enabled && (
+                <Box display={displayProp}>
```
Comment (gomesalexandre) [inline src/components/Layout/Header/Header.tsx:227]: We need a `displayProp2`, see:
```typescript
const displayProp = { base: 'block', md: 'none' }
```
but this should be
```typescript
const displayProp2 = { base: 'none', md: 'block' }
```
```suggestion
                <Box display={displayProp2}>
```

File: src/components/Layout/Header/Header.tsx [hunk 4]
```diff
@@ -186,39 +201,41 @@ export const Header = memo(() => {
...
                   <WalletConnectToDappsHeaderButton />
                 </Box>
               )}
-              <ChainMenu display={{ base: 'none', md: 'block' }} />
+              {isLargerThanMd && <ChainMenu display={displayProp} />}
```
Comment (gomesalexandre) [inline src/components/Layout/Header/Header.tsx:231]: Ditto

File: src/components/Layout/Header/Header.tsx [hunk 5]
```diff
@@ -186,39 +201,41 @@ export const Header = memo(() => {
...
               <TxWindow />
               <Notifications />
-              <Box display={{ base: 'none', md: 'block' }}>
-                <UserMenu />
-              </Box>
+              {isLargerThanMd && (
+                <Box display={displayProp}>
```
Comment (gomesalexandre) [inline src/components/Layout/Header/Header.tsx:235]: And here too

Comment (gomesalexandre) [reply src/components/Layout/Header/Header.tsx]: 47d972a9b7cca742ee17b0bdaec7c9df31095da5

File: src/components/Layout/Header/Header.tsx (hunk 4 above)
Comment (gomesalexandre) [reply src/components/Layout/Header/Header.tsx]: 47d972a9b7cca742ee17b0bdaec7c9df31095da5

File: src/components/Layout/Header/Header.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/components/Layout/Header/Header.tsx]: 47d972a9b7cca742ee17b0bdaec7c9df31095da5

**Review by gomesalexandre — APPROVED** [review]
Tested again locally:

- Mobile nav still looks good

image


- Desktop nav now has user menu, tx history button, as well as chain selection menu displayed with MM connected

image

- Native still displays WC dApps button

image
image

**Pull request merged**